pkgs/top-level/octave-packages.nix

```diff
@@ -0,0 +1,92 @@
+# This file contains the GNU Octave add-on packages set.
+# Each attribute is an Octave library.
+# Expressions for the Octave libraries are supposed to be in `pkgs/development/octave-modules/<name>/default.nix`.
+
+{ pkgs
+, lib
+, stdenv
+, fetchurl
+, newScope
+, octave
+, lapack, blas
+, gfortran
+, autoreconfHook
+, python3
+, python3Packages
+, jdk
+, gnuplot
+, texinfo
+, nettle
+}:
+
+let
+
+  isOctaveFull = octave.enableQt;
+  isOctaveJIT = octave.enableJIT;
+
+  buildOctaveLibrary = callPackage ../development/interpreters/octave/mk-octave-derivation.nix {
+    inherit pkgs lib stdenv;
+    namePrefix = "${octave.pname}-${octave.version}";
+    inherit octave;
+  };
+
+  callPackage = pkgs.newScope {
+    inherit (pkgs) lib stdenv;
+    inherit buildOctaveLibrary;
+    inherit fetchurl;
+    inherit gnuplot texinfo;
+  };
+
+in rec {
+
+  control = callPackage ../development/octave-modules/control {
+    gfortran = gfortran;
+    autoreconfHook = autoreconfHook;
+    lapack = lapack;
+    blas = blas;
```

Comment:
I don't like this, this should be the default - the same `gfortran` that's used for octave should be used here, without question, not choice. For that, you'd need something like this:

https://github.com/NixOS/nixpkgs/commit/2a51ddda6beec6ce04770b140700599d70e253a3

Reply:
I'm sorry, but I'm not quite sure how that commit show-cases what you mean. 
Would you mind explaining the general idea for me? I understand why having a different `gfortran` would be bad, but I'm not sure why manually specifying that `gfortran` is needed for that package is particularly "bad".

Reply:
> I'm sorry, but I'm not quite sure how that commit show-cases what you mean.
> Would you mind explaining the general idea for me? I understand why having a different `gfortran` would be bad, but I'm not sure why manually specifying that `gfortran` is needed for that package is particularly "bad".

The idea is, that if in the future there will be a new octave package added, the person adding it won't have to take care of all of these deps safety issues - that the dependency the octave package will use exactly the same deps (if needed) that octave uses - those that are available in it's passthru. A user adding a new package should do nothing besides `callPackage <file.nix> { };` and whatever deps they wish to use there will be used thanks to the prior definition of `callPackage`.

That commit by `@FRidh` demonstrates how to solve this:

https://github.com/NixOS/nixpkgs/pull/106599#issuecomment-749709846

And essentially it sums up to using `lib.makeScope` instead of `lib.newScope`. I wasn't accurate when I gave the example of `gfortran`, as for this dep specifically we can put it in the scope of `callPackage` even with `newScope` if you do:

```nix
let
  callPackage = lib.newScope {
    # Other deps...
    inherit (octave) gfortran;
  };
in {
  pkg1 = callPackage file.nix { };
  # ...
}
```

But I think we should use `makeScope` never the less, to avoid having to specify deps in usages of `callPackage` as I (hopefully in an understandable manner) explained in https://github.com/NixOS/nixpkgs/pull/106599#issuecomment-749709846 .

Reply:
See https://github.com/NixOS/nixpkgs/pull/108562/commits/a9597730631bb5f8b0086a273811a930cbb7b6a0